code/modules/antagonists/changeling/changeling.dm

```diff
@@ -363,7 +363,7 @@
  *
  * [sting_path] - the power that's being purchased / evolved.
  */
-/datum/antagonist/changeling/proc/purchase_power(datum/action/changeling/sting_path)
+/datum/antagonist/changeling/proc/purchase_power(datum/action/changeling/sting_path, free = FALSE)
```

Comment:
I don't like boolean parameters like this. Instead, Make a new proc named `give_power` that runs all the code, then returns TRUE/FALSE if it succeeded (like this already does now).

Then, have `purchase_power` do something like:

```dm
var/success = give_power(sting_path)
if (success)
    genetic_points -= initial(sting_path.dna_cost)
return success
```